python/ray/tune/result_grid.py

```diff
@@ -13,6 +14,26 @@
 from ray.util import PublicAPI
 
 
+def _convert_dir_checkpoint_to_cloud_checkpoint_if_needed(
+    checkpoint: Checkpoint, trial: Trial
+) -> Checkpoint:
+    """
+    If checkpoint is backed by local dir and trial has remote_checkpoint_dir,
+    return a new URI-backed checkpoint.
+    """
+    if (
+        checkpoint.uri
```

Comment:
The checkpoint might exist on a different node's filesystem, and `checkpoint.uri` would be `None` in that case, since it only returns local path if it exists. But we should still return a URI-backed checkpoint.

We can construct it using `checkpoint._local_path` and `trial.remote_checkpoint_dir` instead.

Q: Should `checkpoint._local_path` be exposed as a public attribute similar to `_uri`?

Reply:
Good point. We can use `get_internal_representation` here instead.